onionshare/settings.py

```diff
@@ -81,6 +82,19 @@ def fill_in_defaults(self):
             if key not in self._settings:
                 self._settings[key] = self.default_settings[key]
 
+    def build_state_dir(self):
+        """
+        Returns the path of the directory that holds Tor state/cache data.
+        """
+        p = platform.system()
+        if p == 'Windows':
+            appdata = os.environ['APPDATA']
+            return '{}\\OnionShare\\tor'.format(appdata)
+        elif p == 'Darwin':
+            return os.path.expanduser('~/Library/Application Support/OnionShare/tor')
+        else:
+            return os.path.expanduser('~/.config/onionshare/tor')
```

Comment:
Is it possible to use the `XDG_CONFIG_HOME` instead?

Following the **XDG Base Directory Specification** these days should be a must for any Linux application.